airbyte-webapp/src/views/Connection/ConnectionForm/ConnectionFormFields.tsx

```diff
@@ -0,0 +1,184 @@
+  const { formatMessage } = useIntl();
+
+  return (
+    <FormContainer className={className}>
+      <Section title={<FormattedMessage id="connection.transfer" />}>
+        <ScheduleField />
+      </Section>
+      <Card>
+        <StyledSection>
+          <H5 bold>
+            <FormattedMessage id="connection.streams" />
+          </H5>
+          <span style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}>
+            <Field name="namespaceDefinition" component={NamespaceDefinitionField} />
+          </span>
+          {values.namespaceDefinition === NamespaceDefinitionType.customformat && (
+            <Field name="namespaceFormat">
+              {({ field, meta }: FieldProps<string>) => (
+                <FlexRow>
+                  <LeftFieldCol>
+                    <NamespaceFormatLabel
+                      nextLine
+                      error={!!meta.error}
+                      label={<FormattedMessage id="connectionForm.namespaceFormat.title" />}
+                      message={<FormattedMessage id="connectionForm.namespaceFormat.subtitle" />}
+                    />
+                  </LeftFieldCol>
+                  <RightFieldCol style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}>
+                    <Input
+                      {...field}
+                      error={!!meta.error}
+                      placeholder={formatMessage({
+                        id: "connectionForm.namespaceFormat.placeholder",
+                      })}
+                    />
+                  </RightFieldCol>
+                </FlexRow>
+              )}
+            </Field>
+          )}
+          <Field name="prefix">
+            {({ field }: FieldProps<string>) => (
+              <FlexRow>
+                <LeftFieldCol>
+                  <ControlLabels
+                    nextLine
+                    label={formatMessage({
+                      id: "form.prefix",
+                    })}
+                    message={formatMessage({
+                      id: "form.prefix.message",
+                    })}
+                  />
+                </LeftFieldCol>
+                <RightFieldCol>
+                  <Input
+                    {...field}
+                    type="text"
+                    placeholder={formatMessage({
+                      id: `form.prefix.placeholder`,
+                    })}
+                    data-testid="prefixInput"
+                    style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}
+                  />
+                </RightFieldCol>
+              </FlexRow>
+            )}
+          </Field>
+        </StyledSection>
+        <StyledSection>
+          <Field
+            name="syncCatalog.streams"
+            component={SchemaField}
+            isSubmitting={isSubmitting}
+            additionalControl={
+              <Button onClick={onRefreshSourceSchema} type="button" secondary>
```

Comment:
Another todo: I wonder if making `RefreshSourceSchema` its own component would buy us anything here in terms of keeping this component as simple as possible on its own.

Reply:
This just leverages Formik-specific logic, I don't think it's worth doing anything more with right now.